test_conformance/extensions/cl_khr_command_buffer/basic_command_buffer.cpp

```diff
@@ -0,0 +1,532 @@
+//
+// Copyright (c) 2021 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "command_buffer_test_base.h"
+#include "procs.h"
+#include "harness/typeWrappers.h"
+
+#include <algorithm>
+#include <cstring>
+#include <vector>
+
+#define CHECK_VERIFICATION_ERROR(reference, result, index)                     \
+    {                                                                          \
+        if (reference != result)                                               \
+        {                                                                      \
+            log_error("Expected %d was %d at index %u\n", reference, result,   \
+                      index);                                                  \
+            return TEST_FAIL;                                                  \
+        }                                                                      \
+    }
+
+namespace {
+
+// Helper test fixture for constructing OpenCL objects used in testing
+// a variety of simple command-buffer enqueue scenarios.
+struct BasicCommandBufferTest : CommandBufferTestBase
+{
+
+    BasicCommandBufferTest(cl_device_id device, cl_context context,
+                           cl_command_queue queue)
+        : CommandBufferTestBase(device), context(context), queue(queue),
+          command_buffer(nullptr), simultaneous_use(false), num_elements(0)
+    {}
+
+    cl_int TearDown()
+    {
+        if (nullptr != command_buffer)
+        {
+            cl_int error = clReleaseCommandBufferKHR(command_buffer);
```

Comment:
We never reach this when a test is skipped, leaking the command buffer.

Reply:
Another good spot, I've moved this release to the test destructor to fix this leak, but ideally we'd want a wrapper for the command-buffer. Discussion - See https://github.com/KhronosGroup/OpenCL-CTS/pull/1368#discussion_r771424219